cmd/lncli/cmd_invoice.go

```diff
@@ -1,14 +1,86 @@
 package main
 
 import (
+	"encoding/base64"
 	"encoding/hex"
+	"encoding/json"
 	"fmt"
 	"strconv"
 
 	"github.com/lightningnetwork/lnd/lnrpc"
 	"github.com/urfave/cli"
 )
 
+// newHopHint creates a new lnrpc.HopHint from the provided parameters.
+func newHopHint(
+	nodeID string, chanID uint64, feeBaseMsat, feeProportionalMillionths,
+	cltvExpiryDelta uint32) (*lnrpc.HopHint, error) {
+
+	return &lnrpc.HopHint{
+		NodeId: nodeID, ChanId: chanID, FeeBaseMsat: feeBaseMsat,
+		FeeProportionalMillionths: feeProportionalMillionths,
+		CltvExpiryDelta:           cltvExpiryDelta,
+	}, nil
+}
+
+func validateAndParseRouteHints(
```

Comment:
naming nit: we are first parsing and then validating

Reply:
actually we arent doing much validation here. So could just be `parseRouteHints` imo